bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetDeviceHandler.java

```diff
@@ -0,0 +1,355 @@
+import java.time.temporal.ChronoField;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.juicenet.internal.api.JuiceNetApi;
+import org.openhab.binding.juicenet.internal.api.JuiceNetApiException;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiCar;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDeviceStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiInfo;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiTouSchedule;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JuiceNetDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetDeviceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetDeviceHandler.class);
+
+    protected final TimeZoneProvider timeZoneProvider;
+
+    // properties
+    protected String id = "";
+    protected String name = "";
+
+    protected String token = "";
+    protected long targetTimeTou = 0;
+    protected long lastInfoTimestamp = 0;
+
+    JuiceNetApiDeviceStatus deviceStatus = new JuiceNetApiDeviceStatus();
+    JuiceNetApiInfo deviceInfo = new JuiceNetApiInfo();
+    JuiceNetApiTouSchedule deviceTouSchedule = new JuiceNetApiTouSchedule();
+    JuiceNetApiCar deviceCar = new JuiceNetApiCar();
+
+    public JuiceNetDeviceHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    public void setNameAndToken(String name, String token) {
+        logger.trace("setNameAndToken");
+        this.token = token;
+
+        if (!name.equals(this.name)) {
+            Map<String, String> properties = editProperties();
+            properties.put(PROPERTY_NAME, name);
+            this.updateProperties(properties);
```

Comment:
There is a convenience method for updating a single property:
```suggestion
            updateProperty(PROPERTY_NAME, name);
```